sdcm/remote/remote_long_running.py

```diff
@@ -0,0 +1,57 @@
+import time
+import logging
+from uuid import uuid4
+from pathlib import Path
+from contextlib import ExitStack
+
+from sdcm.remote.remote_cmd_runner import RemoteCmdRunnerBase
+from sdcm.remote.libssh2_client.result import Result
+from sdcm.remote.libssh2_client.exceptions import UnexpectedExit
+
+logger = logging.getLogger(__name__)
+
+
+def run_long_running_cmd(remoter: RemoteCmdRunnerBase,
+                         cmd: str,
+                         timeout: float | None = None,
+                         ignore_status: bool = False,
+                         verbose: bool = True,
+                         retry: int = 1):
+
+    cmd_uuid = uuid4()
+    cmd_stdout_log = Path('/tmp') / f'remoter_stdout_{cmd_uuid}.log'
+    cmd_stderr_log = Path('/tmp') / f'remoter_stderr_{cmd_uuid}.log'
+    cmd_exit_log = Path('/tmp') / f'remoter_exit_{cmd_uuid}.log'
+    cmd_bash = Path('/tmp') / f'remoter_cmd_{cmd_uuid}.sh'
+
+    with ExitStack() as stack:
+        def clear_tmp_files():
+            for log_file in (cmd_stdout_log, cmd_stderr_log, cmd_exit_log, cmd_bash):
+                remoter.run(f'rm {log_file}', verbose=False, ignore_status=True)
+
+        stack.callback(clear_tmp_files)
+        remoter.run(f'echo "{cmd}" > {cmd_bash}')
+
+        start_time = time.perf_counter()
+
+        remote_command = f'(bash +e +x -o pipefail {cmd_bash} ; echo $? > {cmd_exit_log}) >& {cmd_stdout_log} 2> {cmd_stderr_log} </dev/null  &  jobs -p'
+        if verbose:
+            logger.debug("<%s> execute run_long_running_cmd: %s", remoter.hostname, remote_command)
+        pid = remoter.run(remote_command, retry=0, verbose=verbose).stdout
+
+        remoter.run(f'while [ -e /proc/{pid.strip()} ]; do sleep 0.1; done', timeout=timeout, retry=10, verbose=False)
```

Comment:
if we reach timeout, command is not stopped - this is different than typical way

Reply:
it's not, if we start execution of decommission, it continue even if we timeout or disconnect.
same goes to all of nodetool commands, there no way to stop them once they reached scylla side

Reply:
I think `nodetool` command was stopping (I know decommission is not stopped due `nodetool` interruption)

Reply:
no sure I understood what do you mean here ? 

today if ssh disconnect nodetool itself gonna die, and the command would continue 

but with this code, since we are running it in the background, ssh disconnect won't kill the nodetool command, and this while loop can be killed multiple times, it won't affect the nodetool command

Reply:
I'm just saying that in case of timeout we should kill `pid` - like normal ssh disconnect would do.
Otherwise, if a command hangs indefinitely it will stay running forever.

Reply:
and ? even if we kill it might still be running on scylla end...

it don't think it make much difference at that point, if we really want to cancel such operation, we should be looking into can scylla can provide, it's killing the client isn't enough for most of the cases we are talking about.

Reply:
for these cases not (nodetool) but for others might be problematic - anyway I'm not having now good example, so I'm not insisting. 
Just worth to remember this flaw.